crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -234,6 +248,35 @@ unsafe impl<D: QueryData + 'static, F: QueryFilter + 'static> SystemParam for Qu
     }
 }
 
+impl<'w, 's, D: QueryData + 'static, F: QueryFilter + 'static> BuildableSystemParam
+    for Query<'w, 's, D, F>
+{
+    type Builder<'b> = QueryBuilder<'b, D, F>;
+
+    #[inline]
+    fn build(
+        world: &mut World,
+        system_meta: &mut SystemMeta,
+        build: impl FnOnce(&mut Self::Builder<'_>),
+    ) -> Self::State {
+        let mut builder = QueryBuilder::new(world);
```

Comment:
I think this is all correct, but it looked odd at first so I want to leave a note to make sure I understand correctly.  

`Query::init_state` calls `QueryState::new_with_access` to populate `system_meta.archetype_component_access`.  This function doesn't update `archetype_component_access` at all, so I was worried it wasn't accounting for existing archetypes!  

But `FunctionSystem::update_archetype_component_access` calls `new_archetype` on existing archetypes on the first call, 
so everything will be populated before the system is run.  (And `Query::init_state` is doing unnecessary work by calling `QueryState::new_with_access`, and could just call `QueryState::new_uninitialized`.)